java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/OAuthRefreshTokenSourceAuthStrategy.java

```diff
@@ -209,22 +220,42 @@ private CanonicalOAuthAccessTokenResponseDto exchangeRefreshTokenForAccessToken(
                 objectMapper.readerFor(CanonicalOAuthAccessTokenResponseDto.class)
                     .readValue(response.getContent());
 
-            //TODO: this is obviously not great; if we're going to support refresh token rotation,
-            // need to have some way to control the logic based on grant type
-            config.getConfigPropertyAsOptional(RefreshTokenBuilder.ConfigProperty.REFRESH_TOKEN)
-                .ifPresent(currentRefreshToken -> {
-                    if (!Objects.equals(currentRefreshToken, tokenResponse.getRefreshToken())) {
-                        //TODO: update refreshToken (some source APIs do this; TBC whether ones currently
-                        // in scope for psoxy use do)
-                        //q: write to secret? (most correct ...)
-                        //q: write to file system?
-                        log.severe("Source API rotated refreshToken, which is currently NOT supported by psoxy");
-                    }
-                });
+            storeRefreshTokenIfRotated(tokenResponse);
 
             return tokenResponse;
         }
 
+        /**
+         * Store the new refresh token if included in token response and differs from stored value
+         *
+         * this method encapsulates the logic for checking and storing new token
+         *
+         * @param tokenResponse the token response
+         */
+        @VisibleForTesting
+        void storeRefreshTokenIfRotated(CanonicalOAuthAccessTokenResponseDto tokenResponse) {
+            if (!StringUtils.isBlank(tokenResponse.getRefreshToken())) {
+                //if a refresh_token came back from server, potentially update it
+                config.getConfigPropertyWithMetadata(RefreshTokenTokenRequestBuilder.ConfigProperty.REFRESH_TOKEN)
+                    .filter(storedToken -> !Objects.equals(storedToken.getValue(), tokenResponse.getRefreshToken()))
+                    .filter(storedToken -> storedToken.getLastModifiedDate().isEmpty()
+                        || storedToken.getLastModifiedDate().get()
```

Comment:
are you sure this makes sense??? if a new refresh token comes I'd just assume the old one has been revoked. Id' store it anyway.